stdlib/public/Reflection/TypeLowering.cpp

```diff
@@ -1295,12 +1896,39 @@ class EnumTypeInfoBuilder {
         Alignment = FixedDescriptor->getAlignment();
         NumExtraInhabitants = FixedDescriptor->NumExtraInhabitants;
         BitwiseTakable = FixedDescriptor->isBitwiseTakable();
+        unsigned Stride = ((Size + Alignment - 1) & ~(Alignment - 1));
+        if (Stride == 0)
+          Stride = 1;
+
+        auto PayloadSize = EnumTypeInfo::getPayloadSizeForCases(Cases);
+        BitMask spareBitsMask(PayloadSize);
+        auto validSpareBitsMask = populateSpareBitsMask(Cases, spareBitsMask);
+
+        if (!validSpareBitsMask) {
+          // If we couldn't correctly determine the spare bits mask,
+          // return a TI that will always fail when asked for XIs or value.
+          return TC.makeTypeInfo<UnsupportedEnumTypeInfo>(
+            Size, Alignment, Stride, NumExtraInhabitants,
+            BitwiseTakable, EnumKind::MultiPayloadEnum, Cases);
+        } else if (spareBitsMask.isZero()) {
+          // Simple case that does not use spare bits
+          // This is correct as long as our local spare bits calculation
+          // above only returns an empty mask when the mask is really empty,
+          return TC.makeTypeInfo<SimpleMultiPayloadEnumTypeInfo>(
```

Comment:
If I were to change this (and/or the case immediately below) to use `UnsupportedEnumTypeInfo`, I believe that would suffice to make this PR mergeable.  Some MPEs would not be projectable then, but that's inevitable even if we did make compiler changes, as this library will sometimes get used to inspect old binaries.  The important thing is that anything this library does project gets projected correctly.

Reply:
Fixed-layout MPEs are now `Unsupported` so they won't be projected by the current code.  After I can add some mechanism for the compiler to expose accurate spare bit information, this can be revisited.